libkbfs/prefetcher.go

```diff
@@ -319,7 +319,11 @@ func (p *blockPrefetcher) request(ctx context.Context, priority int,
 			NoPrefetch, isDeepSync, nil}
 		pre = p.newPrefetch(1, false, req)
 		p.prefetches[ptr.ID] = pre
-		ch := p.retriever.Request(pre.ctx, priority, kmd, ptr, block, lifetime)
+		requester := p.retriever.Request
+		if isDeepSync {
+			requester = p.retriever.RequestAndSync
+		}
+		ch := requester(pre.ctx, priority, kmd, ptr, block, lifetime)
```

Comment:
The code structure would be the same, except it would be setting an action variable in the `if` instead of the method.  Seems the same to me, but I'll leave it up to you to make the call about changing it.  Let me know.